src/main/java/seedu/duke/ui/UI.java

```diff
@@ -159,4 +183,87 @@ public void showInvalidStepMessage() {
     public void showErrorMessage(Exception e) {
         System.out.println(e.getMessage());
     }
+    public boolean isValidIntegerInputToAdd(String input, int maxSteps) {
+        try {
+            int integerInput = Integer.parseInt(input.trim());
+            if (integerInput < 1 || integerInput > maxSteps + 1) {
+                return false;
+            }
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+    public boolean isValidIntegerInputToDelete(String input, int maxSteps) {
+        try {
+            int integerInput = Integer.parseInt(input.trim());
+            if (integerInput < 1 || integerInput > maxSteps) {
+                return false;
+            }
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+    public void showInvalidIndexMessage() {
+        System.out.println(StringLib.INVALID_STEP_INDEX);
+    }
+    public void requestIndexInput() {
+        System.out.println(StringLib.INDEX_REQUEST);
+    }
+    public int getIndexToAdd(int maxSteps) {
+        System.out.println(StringLib.INDEX_REQUEST);
+        String userInput = in.nextLine();
+        while (!isValidIntegerInputToAdd(userInput, maxSteps)) {
+            showInvalidIndexMessage();
+            System.out.println("Valid range: " + 1 + " to " + (maxSteps + 1));
```

Comment:
@YatPang perhaps the `maxSteps + 1` is the issue that is causing the valid range maximum to be incorrect.

Reply:
Non-issue: Command allows new step to be added as the final step on the step list.
Considered resolved.